src/retry.rs

```diff
@@ -0,0 +1,25 @@
+use tokio_retry::strategy::{FibonacciBackoff, jitter};
+use std::future::Future;
+use std::{
+  iter::{Take, Map},
+  time::Duration
+};
+
+fn retry_strategy(ms: u64, attempts: usize) -> Take<Map<FibonacciBackoff, fn(Duration) -> Duration>>{
+  FibonacciBackoff::from_millis(ms)
+    .map(jitter as fn(Duration) -> Duration)
+    .take(attempts)
+}
+
+
+pub async fn with_retry<A, F, R, E>(ms: Option<u64>, attempts: Option<usize>, action: A) -> Result<R, E>
+  where
+    A: FnMut() -> F,
+    E: std::fmt::Display,
```

Comment:
This bound probably isn't necessary. The tokio-retry crate does not require the error type to implement `Display` and the function itself does not format/display the error.